auction-app-backend/src/main/java/com/example/auctionapp/service/implementation/ProductServiceImpl.java

```diff
@@ -29,11 +34,65 @@ public ProductServiceImpl(final ProductRepository productRepository, final Categ
     }
 
     @Override
-    public Page<Product> getProducts(final UUID categoryId, final String searchProduct, final int page, final int size) {
+    public Map<String, Object> getProducts(UUID categoryId, String searchProduct, int page, int size) {
         Pageable pageable = PageRequest.of(page, size);
         Specification<ProductEntity> specification = ProductSpecification.withDynamicQuery(categoryId, searchProduct);
 
-        return productRepository.findAll(specification, pageable).map(ProductEntity::toDomainModel);
+        Page<Product> products = productRepository.findAll(specification, pageable).map(ProductEntity::toDomainModel);
+        Map<String, Object> response = new HashMap<>();
```

Comment:
Can we make an object for this response, that will consists of string and Product and you just set those values instead of using HashMap